Click on a PVC results in a stacktrace

**What steps did you take and what happened:**
I used octant to browse my cluster.  When I clicked on a PVC (specifically I clicked on the **Volume** field of the PVC)  I got a stacktrace:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x2129a21]

goroutine 212 [running]:
github.com/vmware-tanzu/octant/internal/printer.(*PersistentVolumeConfiguration).Create(0xc0009bc550, 0x0, 0x3bca540, 0xc000f42680, 0x3baa9c0, 0xc000a4c040, 0x2125804, 0x2516700, 0xc0010c2800)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:163 +0x6e1
github.com/vmware-tanzu/octant/internal/printer.defaultPersistentVolumeConfig(0xc000140a00, 0x0, 0x3bca540, 0xc000f42680, 0x3baa9c0, 0xc000a4c040, 0x245eea0, 0xc00024a880, 0x0)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:127 +0x63
github.com/vmware-tanzu/octant/internal/printer.(*persistentVolumeHandler).Config(0xc00024a880, 0x0, 0x3bca540, 0xc000f42680, 0x3baa9c0, 0xc000a4c040, 0x2415800, 0xc000f0fa80)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:109 +0x54
github.com/vmware-tanzu/octant/internal/printer.PersistentVolumeHandler(0x3baa600, 0xc000f0fa80, 0xc000140a00, 0x0, 0x3bca540, 0xc000f42680, 0x3baa9c0, 0xc000a4c040, 0x0, 0x0, ...)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/printer/persistentvolume.go:72 +0xc1
reflect.Value.call(0x236b920, 0x39a0168, 0x13, 0x258603e, 0x4, 0xc0009bcab8, 0x3, 0x3, 0x100bebf, 0x2421140, ...)
	/usr/local/Cellar/go/1.14.1/libexec/src/reflect/value.go:460 +0x8ab
reflect.Value.Call(0x236b920, 0x39a0168, 0x13, 0xc0009bcab8, 0x3, 0x3, 0x10198de, 0x4fa76d0, 0x0)
	/usr/local/Cellar/go/1.14.1/libexec/src/reflect/value.go:321 +0xb4
github.com/vmware-tanzu/octant/internal/printer.(*Resource).Print(0xc00024a080, 0x3baa600, 0xc000f0fa80, 0x3b7b300, 0xc000140a00, 0x3baaa00, 0xc0004f6630, 0x23ea3c0, 0xc0009bcbc8, 0xc0009bcbf8, ...)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/printer/printer.go:77 +0x472
github.com/vmware-tanzu/octant/internal/describer.(*Object).addSummaryTab(0xc000f872c0, 0x3baa600, 0xc000f0fa80, 0x3b7b300, 0xc000140a00, 0xc00018e8c0, 0x3bca540, 0xc000f42680, 0x3bc55a0, 0xc000b30040, ...)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/describer/object.go:214 +0xab
github.com/vmware-tanzu/octant/internal/describer.(*Object).Describe(0xc000f872c0, 0x3baa600, 0xc000f0fa80, 0x0, 0x0, 0x3bca540, 0xc000f42680, 0x3bc55a0, 0xc000b30040, 0xc000c7a150, ...)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/describer/object.go:154 +0xf52
github.com/vmware-tanzu/octant/internal/modules/clusteroverview.(*ClusterOverview).Content(0xc0010c5080, 0x3baa600, 0xc000f0fa80, 0xc0008f4420, 0x3c, 0xc000a4c010, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/modules/clusteroverview/clusteroverview.go:167 +0x750
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).generateContent(0xc000f71440, 0x3baa600, 0xc000f0fa80, 0x3bc91e0, 0xc00015ac60, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:161 +0x320
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).runUpdate.func1(0x3baa600, 0xc000f0fa80, 0x485b560)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:115 +0xe0
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run.func1(0x3baa600, 0xc000f71400, 0x3bbae80, 0xc000010a10, 0xc0009bde77, 0xc000f015c0, 0xc0010b2050, 0xc000810390, 0x3b9aca00)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/api/poller.go:86 +0x271
github.com/vmware-tanzu/octant/internal/api.(*InterruptiblePoller).Run(0xc001220b30, 0x3baa600, 0xc000f71400, 0xc000f015c0, 0xc000810390, 0x3b9aca00)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/api/poller.go:95 +0x2ff
github.com/vmware-tanzu/octant/internal/api.(*ContentManager).Start(0xc000f71440, 0x3baa600, 0xc000f71400, 0x3bc91e0, 0xc00015ac60, 0x3b78cc0, 0xc000272d80)
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/api/content_manager.go:105 +0x1c7
created by github.com/vmware-tanzu/octant/internal/api.(*WebsocketState).Start
	/private/tmp/octant-20200330-46047-1cpjkwt/src/github.com/vmware-tanzu/octant/internal/api/websocket_state.go:142 +0x9b
```


**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]


**Environment:**

- Octant version (use `octant version`): 
Version:  0.11.1
Git commit:  a5f7847a7e21d781fdb7edd5389b04edc2ca6b87
Built:  2020-03-30T19:51:23Z

- Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.4", GitCommit:"8d8aa39598534325ad77120c120a22b3a990b5ea", GitTreeState:"clean", BuildDate:"2020-03-12T23:41:24Z", GoVersion:"go1.14", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.5", GitCommit:"20c265fef0741dd71a66480e35bd69f18351daea", GitTreeState:"clean", BuildDate:"2019-10-15T19:07:57Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

- OS (macOS 10.15, Windows 10, Ubuntu 19.10 etc):

macOS 10.5.3

--- a/changelogs/unreleased/817-GuessWhoSamFoo
+++ b/changelogs/unreleased/817-GuessWhoSamFoo
@@ -1,0 +1,1 @@
+Added support for different persistent volume sources in summary view

--- a/internal/printer/persistentvolume.go
+++ b/internal/printer/persistentvolume.go
@@ -10,7 +10,7 @@ import (
 	"fmt"
 
 	"github.com/pkg/errors"
-	v1 "k8s.io/api/core/v1"
+	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 
 	"github.com/vmware-tanzu/octant/pkg/store"
@@ -18,7 +18,7 @@ import (
 )
 
 // PersistentVolumeListHandler is a printFunc that creates a component to display multiple Persistent Volumes
-func PersistentVolumeListHandler(ctx context.Context, list *v1.PersistentVolumeList, options Options) (component.Component, error) {
+func PersistentVolumeListHandler(ctx context.Context, list *corev1.PersistentVolumeList, options Options) (component.Component, error) {
 	if list == nil {
 		return nil, errors.New("nil list")
 	}
@@ -33,7 +33,7 @@ func PersistentVolumeListHandler(ctx context.Context, list *v1.PersistentVolumeL
 			return nil, err
 		}
 
-		storage := pv.Spec.Capacity[v1.ResourceStorage]
+		storage := pv.Spec.Capacity[corev1.ResourceStorage]
 		capacity := storage.String()
 
 		accessModes := getAccessModesAsString(pv.Spec.AccessModes)
@@ -60,7 +60,7 @@ func PersistentVolumeListHandler(ctx context.Context, list *v1.PersistentVolumeL
 }
 
 // PersistentVolumeHandler is a printFunc that creates a component to display a single Persistent Volume
-func PersistentVolumeHandler(ctx context.Context, pv *v1.PersistentVolume, options Options) (component.Component, error) {
+func PersistentVolumeHandler(ctx context.Context, pv *corev1.PersistentVolume, options Options) (component.Component, error) {
 	obj := NewObject(pv)
 	obj.EnableEvents()
 
@@ -81,13 +81,13 @@ func PersistentVolumeHandler(ctx context.Context, pv *v1.PersistentVolume, optio
 }
 
 type persistentVolumeHandler struct {
-	configFunc       func(*v1.PersistentVolume, Options) (*component.Summary, error)
-	statusFunc       func(context.Context, *v1.PersistentVolume, Options) (*component.Summary, error)
-	persistentVolume *v1.PersistentVolume
+	configFunc       func(*corev1.PersistentVolume, Options) (*component.Summary, error)
+	statusFunc       func(context.Context, *corev1.PersistentVolume, Options) (*component.Summary, error)
+	persistentVolume *corev1.PersistentVolume
 	object           *Object
 }
 
-func newPersistentVolumeHandler(pv *v1.PersistentVolume, object *Object) (*persistentVolumeHandler, error) {
+func newPersistentVolumeHandler(pv *corev1.PersistentVolume, object *Object) (*persistentVolumeHandler, error) {
 	if pv == nil {
 		return nil, errors.New("cannot print a nil persistentVolume")
 	}
@@ -123,22 +123,22 @@ func (pvh *persistentVolumeHandler) Status(ctx context.Context, options Options)
 	return nil
 }
 
-func defaultPersistentVolumeConfig(pv *v1.PersistentVolume, options Options) (*component.Summary, error) {
+func defaultPersistentVolumeConfig(pv *corev1.PersistentVolume, options Options) (*component.Summary, error) {
 	return NewPersistentVolumeConfiguration(pv).Create(options)
 }
 
-func defaultPersistentVolumeStatus(ctx context.Context, pv *v1.PersistentVolume, options Options) (*component.Summary, error) {
+func defaultPersistentVolumeStatus(ctx context.Context, pv *corev1.PersistentVolume, options Options) (*component.Summary, error) {
 	return NewPersistentVolumeStatus(pv).Create(ctx, options)
 }
 
 // PersistentVolumeConfiguration is used to create the Persistent Volume's configuration component
 // when displaying a single Persistent Volume
 type PersistentVolumeConfiguration struct {
-	persistentVolume *v1.PersistentVolume
+	persistentVolume *corev1.PersistentVolume
 }
 
 // NewPersistentVolumeConfiguration creates a new PersistentVolumeConfiguration using the supplied Persistent Volume
-func NewPersistentVolumeConfiguration(pv *v1.PersistentVolume) *PersistentVolumeConfiguration {
+func NewPersistentVolumeConfiguration(pv *corev1.PersistentVolume) *PersistentVolumeConfiguration {
 	return &PersistentVolumeConfiguration{
 		persistentVolume: pv,
 	}
@@ -152,7 +152,7 @@ func (pvc *PersistentVolumeConfiguration) Create(options Options) (*component.Su
 	pv := pvc.persistentVolume
 
 	accessModes := getAccessModesAsString(pv.Spec.AccessModes)
-	storage := pv.Spec.Capacity[v1.ResourceStorage]
+	storage := pv.Spec.Capacity[corev1.ResourceStorage]
 	capacity := storage.String()
 
 	var sections component.SummarySections
@@ -160,7 +160,11 @@ func (pvc *PersistentVolumeConfiguration) Create(options Options) (*component.Su
 	sections.AddText("Storage Class", pv.Spec.StorageClassName)
 	sections.AddText("Access Modes", accessModes)
 	sections.AddText("Capacity", capacity)
-	sections.AddText("Host Path", pv.Spec.HostPath.Path)
+
+	sections, err := printPersistentVolumeSource(pv, sections)
+	if err != nil {
+		return nil, err
+	}
 
 	summary := component.NewSummary("Configuration", sections...)
 	return summary, nil
@@ -169,11 +173,11 @@ func (pvc *PersistentVolumeConfiguration) Create(options Options) (*component.Su
 // PersistentVolumeStatus is used to create the Persistent Volume's status component
 // when displaying a single Persistent Volume
 type PersistentVolumeStatus struct {
-	persistentVolume *v1.PersistentVolume
+	persistentVolume *corev1.PersistentVolume
 }
 
 // NewPersistentVolumeStatus creates a new PersistentVolumeStatus using the supplied Persistent Volume
-func NewPersistentVolumeStatus(pv *v1.PersistentVolume) *PersistentVolumeStatus {
+func NewPersistentVolumeStatus(pv *corev1.PersistentVolume) *PersistentVolumeStatus {
 	return &PersistentVolumeStatus{
 		persistentVolume: pv,
 	}
@@ -204,9 +208,9 @@ func (pvs *PersistentVolumeStatus) Create(ctx context.Context, options Options) 
 	return summary, nil
 }
 
-func getBoundPersistentVolumeClaim(ctx context.Context, pv *v1.PersistentVolume, options Options) (*v1.PersistentVolumeClaim, error) {
+func getBoundPersistentVolumeClaim(ctx context.Context, pv *corev1.PersistentVolume, options Options) (*corev1.PersistentVolumeClaim, error) {
 	objectStore := options.DashConfig.ObjectStore()
-	pvc := &v1.PersistentVolumeClaim{}
+	pvc := &corev1.PersistentVolumeClaim{}
 
 	cr := pv.Spec.ClaimRef
 	key := store.Key{
@@ -233,7 +237,7 @@ func getBoundPersistentVolumeClaim(ctx context.Context, pv *v1.PersistentVolume,
 	return pvc, nil
 }
 
-func createBoundPersistentVolumeClaimLink(ctx context.Context, pv *v1.PersistentVolume, options Options) (*component.Link, error) {
+func createBoundPersistentVolumeClaimLink(ctx context.Context, pv *corev1.PersistentVolume, options Options) (*component.Link, error) {
 	pvc, err := getBoundPersistentVolumeClaim(ctx, pv, options)
 	if err != nil {
 		return nil, err
@@ -247,4 +251,56 @@ func createBoundPersistentVolumeClaimLink(ctx context.Context, pv *v1.Persistent
 	}
 
 	return claimLink, nil
+}
+
+func printPersistentVolumeSource(pv *corev1.PersistentVolume, section component.SummarySections) (component.SummarySections, error) {
+	switch {
+	case pv.Spec.PersistentVolumeSource.GCEPersistentDisk != nil:
+		section.AddText("GCE Persistent Disk", describeVolumeSource(pv.Spec.PersistentVolumeSource.GCEPersistentDisk))
+	case pv.Spec.PersistentVolumeSource.AWSElasticBlockStore != nil:
+		section.AddText("AWS Elastic Block Store", describeVolumeSource(pv.Spec.PersistentVolumeSource.AWSElasticBlockStore))
+	case pv.Spec.PersistentVolumeSource.HostPath != nil:
+		section.AddText("Host Path", describeVolumeSource(pv.Spec.PersistentVolumeSource.HostPath))
+	case pv.Spec.PersistentVolumeSource.Glusterfs != nil:
+		section.AddText("GlusterFS", describeVolumeSource(pv.Spec.PersistentVolumeSource.Glusterfs))
+	case pv.Spec.PersistentVolumeSource.NFS != nil:
+		section.AddText("NFS", describeVolumeSource(pv.Spec.PersistentVolumeSource.NFS))
+	case pv.Spec.PersistentVolumeSource.RBD != nil:
+		section.AddText("RBD", describeVolumeSource(pv.Spec.PersistentVolumeSource.RBD))
+	case pv.Spec.PersistentVolumeSource.ISCSI != nil:
+		section.AddText("ISCI", describeVolumeSource(pv.Spec.PersistentVolumeSource.ISCSI))
+	case pv.Spec.PersistentVolumeSource.Cinder != nil:
+		section.AddText("Cinder", describeVolumeSource(pv.Spec.PersistentVolumeSource.Cinder))
+	case pv.Spec.PersistentVolumeSource.CephFS != nil:
+		section.AddText("CephFS", describeVolumeSource(pv.Spec.PersistentVolumeSource.CephFS))
+	case pv.Spec.PersistentVolumeSource.FC != nil:
+		section.AddText("FC", describeVolumeSource(pv.Spec.PersistentVolumeSource.FC))
+	case pv.Spec.PersistentVolumeSource.Flocker != nil:
+		section.AddText("Flocker", describeVolumeSource(pv.Spec.PersistentVolumeSource.Flocker))
+	case pv.Spec.PersistentVolumeSource.FlexVolume != nil:
+		section.AddText("Flex Volume", describeVolumeSource(pv.Spec.PersistentVolumeSource.FlexVolume))
+	case pv.Spec.PersistentVolumeSource.AzureFile != nil:
+		section.AddText("Azure File", describeVolumeSource(pv.Spec.PersistentVolumeSource.AzureFile))
+	case pv.Spec.PersistentVolumeSource.VsphereVolume != nil:
+		section.AddText("Vsphere Volume", describeVolumeSource(pv.Spec.PersistentVolumeSource.VsphereVolume))
+	case pv.Spec.PersistentVolumeSource.Quobyte != nil:
+		section.AddText("Quobyte", describeVolumeSource(pv.Spec.PersistentVolumeSource.Quobyte))
+	case pv.Spec.PersistentVolumeSource.AzureDisk != nil:
+		section.AddText("Azure Disk", describeVolumeSource(pv.Spec.PersistentVolumeSource.AzureDisk))
+	case pv.Spec.PersistentVolumeSource.PhotonPersistentDisk != nil:
+		section.AddText("Photon Persistent Disk", describeVolumeSource(pv.Spec.PersistentVolumeSource.PhotonPersistentDisk))
+	case pv.Spec.PersistentVolumeSource.PortworxVolume != nil:
+		section.AddText("Portworx Volume", describeVolumeSource(pv.Spec.PersistentVolumeSource.PortworxVolume))
+	case pv.Spec.PersistentVolumeSource.ScaleIO != nil:
+		section.AddText("ScaleIO", describeVolumeSource(pv.Spec.PersistentVolumeSource.ScaleIO))
+	case pv.Spec.PersistentVolumeSource.Local != nil:
+		section.AddText("Local", describeVolumeSource(pv.Spec.PersistentVolumeSource.Local))
+	case pv.Spec.PersistentVolumeSource.StorageOS != nil:
+		section.AddText("StorageOS", describeVolumeSource(pv.Spec.PersistentVolumeSource.StorageOS))
+	case pv.Spec.PersistentVolumeSource.CSI != nil:
+		section.AddText("CSI", describeVolumeSource(pv.Spec.PersistentVolumeSource.CSI))
+	default:
+		section.AddText("Persistent Volume Source", "Unknown")
+	}
+	return section, nil
 }

--- a/internal/printer/persistentvolume_test.go
+++ b/internal/printer/persistentvolume_test.go
@@ -1,0 +1,605 @@
+/*
+Copyright (c) 2020 the Octant contributors. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package printer
+
+import (
+	"context"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+	"github.com/vmware-tanzu/octant/internal/testutil"
+	"github.com/vmware-tanzu/octant/pkg/store"
+	"github.com/vmware-tanzu/octant/pkg/view/component"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_PersistentVolumeListHandler(t *testing.T) {
+	cols := component.NewTableCols("Name", "Capacity", "Access Modes", "Reclaim Policy", "Status", "Claim", "Storage Class", "Reason", "Age")
+	now := testutil.Time()
+
+	pvcObject := testutil.CreatePersistentVolumeClaim("pvc")
+
+	object := testutil.CreatePersistentVolume("persistentVolume")
+	object.CreationTimestamp = metav1.Time{Time: now}
+	object.Spec.ClaimRef = &corev1.ObjectReference{
+		APIVersion: pvcObject.APIVersion,
+		Kind:       pvcObject.Kind,
+		Name:       pvcObject.Name,
+		Namespace:  pvcObject.Namespace,
+	}
+
+	list := &corev1.PersistentVolumeList{
+		Items: []corev1.PersistentVolume{*object},
+	}
+
+	cases := []struct {
+		name     string
+		list     *corev1.PersistentVolumeList
+		expected *component.Table
+		isErr    bool
+	}{
+		{
+			name: "in general",
+			list: list,
+			expected: component.NewTableWithRows("Persistent Volumes", "We couldn't find any persistent volumes!", cols,
+				[]component.TableRow{
+					{
+						"Name":           component.NewLink("", "persistentVolume", "/persistentVolume"),
+						"Capacity":       component.NewText("0"),
+						"Access Modes":   component.NewText(""),
+						"Reclaim Policy": component.NewText(""),
+						"Status":         component.NewText("Bound"),
+						"Claim":          component.NewLink("", "namespace/pvc", "/pvc"),
+						"Storage Class":  component.NewText(""),
+						"Reason":         component.NewText(""),
+						"Age":            component.NewTimestamp(now),
+					},
+				}),
+		},
+		{
+			name:  "list is nil",
+			list:  nil,
+			isErr: true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			tpo := newTestPrinterOptions(controller)
+			printOptions := tpo.ToOptions()
+
+			ctx := context.Background()
+
+			if tc.list != nil {
+				tpo.PathForObject(&tc.list.Items[0], tc.list.Items[0].Name, "/persistentVolume")
+
+				pvcKey, err := store.KeyFromObject(pvcObject)
+				require.NoError(t, err)
+
+				tpo.objectStore.EXPECT().Get(ctx, pvcKey).
+					Return(testutil.ToUnstructured(t, pvcObject), nil)
+
+				tpo.PathForObject(pvcObject, pvcObject.Namespace+"/"+pvcObject.Name, "/pvc")
+			}
+			got, err := PersistentVolumeListHandler(ctx, tc.list, printOptions)
+			if tc.isErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+
+			component.AssertEqual(t, tc.expected, got)
+		})
+	}
+}
+
+func Test_PersistentVolumeConfiguration(t *testing.T) {
+	persistentVolume := testutil.CreatePersistentVolume("persistentVolume")
+	persistentVolume.Spec.PersistentVolumeSource = corev1.PersistentVolumeSource{
+		HostPath: &corev1.HostPathVolumeSource{
+			Path: "/private/tmp",
+		},
+	}
+
+	cases := []struct {
+		name             string
+		persistentVolume *corev1.PersistentVolume
+		expected         component.Component
+		isErr            bool
+	}{
+		{
+			name:             "host path",
+			persistentVolume: persistentVolume,
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Reclaim Policy",
+					Content: component.NewText(""),
+				},
+				{
+					Header:  "Storage Class",
+					Content: component.NewText(""),
+				},
+				{
+					Header:  "Access Modes",
+					Content: component.NewText(""),
+				},
+				{
+					Header:  "Capacity",
+					Content: component.NewText("0"),
+				},
+				{
+					Header:  "Host Path",
+					Content: component.NewText("{\"path\":\"/private/tmp\"}"),
+				},
+			}...),
+		},
+		{
+			name:             "nil persistentVolume",
+			persistentVolume: nil,
+			isErr:            true,
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			controller := gomock.NewController(t)
+			defer controller.Finish()
+
+			tpo := newTestPrinterOptions(controller)
+			printOptions := tpo.ToOptions()
+
+			pc := NewPersistentVolumeConfiguration(tc.persistentVolume)
+
+			summary, err := pc.Create(printOptions)
+			if tc.isErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+
+			component.AssertEqual(t, tc.expected, summary)
+		})
+	}
+}
+
+func Test_printPersistentVolumeSource(t *testing.T) {
+	cases := []struct {
+		name             string
+		persistentVolume corev1.PersistentVolume
+		expected         *component.Summary
+	}{
+		{
+			name: "gce persistent disk",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						GCEPersistentDisk: &corev1.GCEPersistentDiskVolumeSource{
+							PDName: "my-data-disk",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "GCE Persistent Disk",
+					Content: component.NewText("{\"pdName\":\"my-data-disk\"}"),
+				},
+			}...),
+		},
+		{
+			name: "aws ebs",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						AWSElasticBlockStore: &corev1.AWSElasticBlockStoreVolumeSource{
+							VolumeID: "vol-0b402cb854e070fad",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "AWS Elastic Block Store",
+					Content: component.NewText("{\"volumeID\":\"vol-0b402cb854e070fad\"}"),
+				},
+			}...),
+		},
+		{
+			name: "host path",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: "/private/tmp",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Host Path",
+					Content: component.NewText("{\"path\":\"/private/tmp\"}"),
+				},
+			}...),
+		},
+		{
+			name: "glusterfs",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						Glusterfs: &corev1.GlusterfsPersistentVolumeSource{
+							EndpointsName: "192.168.122.221:1",
+							Path:          "kube_vol",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "GlusterFS",
+					Content: component.NewText("{\"endpoints\":\"192.168.122.221:1\",\"path\":\"kube_vol\"}"),
+				},
+			}...),
+		},
+		{
+			name: "nfs",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						NFS: &corev1.NFSVolumeSource{
+							Server: "nfs.example.com",
+							Path:   "/share1",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "NFS",
+					Content: component.NewText("{\"server\":\"nfs.example.com\",\"path\":\"/share1\"}"),
+				},
+			}...),
+		},
+		{
+			name: "rbd",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						RBD: &corev1.RBDPersistentVolumeSource{
+							CephMonitors: []string{"10.16.154.78:6789", "10.16.154.82:6789"},
+							RBDImage:     "foo",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "RBD",
+					Content: component.NewText("{\"monitors\":[\"10.16.154.78:6789\",\"10.16.154.82:6789\"],\"image\":\"foo\"}"),
+				},
+			}...),
+		},
+		{
+			name: "isci",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						ISCSI: &corev1.ISCSIPersistentVolumeSource{
+							TargetPortal: "10.0.2.15:3260",
+							IQN:          "iqn.2001-04.com.example:storage.kube.sys1.xyz",
+							Lun:          0,
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "ISCI",
+					Content: component.NewText("{\"targetPortal\":\"10.0.2.15:3260\",\"iqn\":\"iqn.2001-04.com.example:storage.kube.sys1.xyz\",\"lun\":0}"),
+				},
+			}...),
+		},
+		{
+			name: "cinder",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						Cinder: &corev1.CinderPersistentVolumeSource{
+							VolumeID: "573e024d-5235-49ce-8332-be1576d323f8",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Cinder",
+					Content: component.NewText("{\"volumeID\":\"573e024d-5235-49ce-8332-be1576d323f8\"}"),
+				},
+			}...),
+		},
+		{
+			name: "cephfs",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						CephFS: &corev1.CephFSPersistentVolumeSource{
+							Monitors: []string{"10.16.154.78:6789", "10.16.154.82:6789"},
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "CephFS",
+					Content: component.NewText("{\"monitors\":[\"10.16.154.78:6789\",\"10.16.154.82:6789\"]}"),
+				},
+			}...),
+		},
+		{
+			name: "fc",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						FC: &corev1.FCVolumeSource{
+							FSType: "ext4",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "FC",
+					Content: component.NewText("{\"fsType\":\"ext4\"}"),
+				},
+			}...),
+		},
+		{
+			name: "flocker",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						Flocker: &corev1.FlockerVolumeSource{
+							DatasetName: "my-flocker-vol",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Flocker",
+					Content: component.NewText("{\"datasetName\":\"my-flocker-vol\"}"),
+				},
+			}...),
+		},
+		{
+			name: "flex volume",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						FlexVolume: &corev1.FlexPersistentVolumeSource{
+							Driver: "kubernetes.io/lvm",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Flex Volume",
+					Content: component.NewText("{\"driver\":\"kubernetes.io/lvm\"}"),
+				},
+			}...),
+		},
+		{
+			name: "azure file",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						AzureFile: &corev1.AzureFilePersistentVolumeSource{
+							SecretName: "azure-secret",
+							ShareName:  "k8stest",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Azure File",
+					Content: component.NewText("{\"secretName\":\"azure-secret\",\"shareName\":\"k8stest\",\"secretNamespace\":null}"),
+				},
+			}...),
+		},
+		{
+			name: "vsphere volume",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						VsphereVolume: &corev1.VsphereVirtualDiskVolumeSource{
+							VolumePath: "/private/tmp",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Vsphere Volume",
+					Content: component.NewText("{\"volumePath\":\"/private/tmp\"}"),
+				},
+			}...),
+		},
+		{
+			name: "quobyte",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						Quobyte: &corev1.QuobyteVolumeSource{
+							Registry: "registry:7861",
+							Volume:   "testVolume",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Quobyte",
+					Content: component.NewText("{\"registry\":\"registry:7861\",\"volume\":\"testVolume\"}"),
+				},
+			}...),
+		},
+		{
+			name: "azure disk",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						AzureDisk: &corev1.AzureDiskVolumeSource{
+							DiskName:    "test.vhd",
+							DataDiskURI: "https://someaccount.blob.microsoft.net/vhds/test.vhd",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Azure Disk",
+					Content: component.NewText("{\"diskName\":\"test.vhd\",\"diskURI\":\"https://someaccount.blob.microsoft.net/vhds/test.vhd\"}"),
+				},
+			}...),
+		},
+		{
+			name: "photon persistent disk",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						PhotonPersistentDisk: &corev1.PhotonPersistentDiskVolumeSource{
+							PdID:   "e8dbc38c-9374-4099-8a09-fcf4be0a7641",
+							FSType: "ext4",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Photon Persistent Disk",
+					Content: component.NewText("{\"pdID\":\"e8dbc38c-9374-4099-8a09-fcf4be0a7641\",\"fsType\":\"ext4\"}"),
+				},
+			}...),
+		},
+		{
+			name: "portworx",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						PortworxVolume: &corev1.PortworxVolumeSource{
+							VolumeID: "vol1",
+							FSType:   "ext4",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Portworx Volume",
+					Content: component.NewText("{\"volumeID\":\"vol1\",\"fsType\":\"ext4\"}"),
+				},
+			}...),
+		},
+		{
+			name: "scaleio",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						ScaleIO: &corev1.ScaleIOPersistentVolumeSource{
+							Gateway:    "https://localhost:443/api",
+							System:     "scaleio",
+							VolumeName: "vol-0",
+							SecretRef: &corev1.SecretReference{
+								Name: "sio-secret",
+							},
+							FSType: "xfs",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "ScaleIO",
+					Content: component.NewText("{\"gateway\":\"https://localhost:443/api\",\"system\":\"scaleio\",\"secretRef\":{\"name\":\"sio-secret\"},\"volumeName\":\"vol-0\",\"fsType\":\"xfs\"}"),
+				},
+			}...),
+		},
+		{
+			name: "local",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						Local: &corev1.LocalVolumeSource{
+							Path: "/private/tmp",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "Local",
+					Content: component.NewText("{\"path\":\"/private/tmp\"}"),
+				},
+			}...),
+		},
+		{
+			name: "storageos",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						StorageOS: &corev1.StorageOSPersistentVolumeSource{
+							VolumeName: "vol-0",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "StorageOS",
+					Content: component.NewText("{\"volumeName\":\"vol-0\"}"),
+				},
+			}...),
+		},
+		{
+			name: "csi",
+			persistentVolume: corev1.PersistentVolume{
+				Spec: corev1.PersistentVolumeSpec{
+					PersistentVolumeSource: corev1.PersistentVolumeSource{
+						CSI: &corev1.CSIPersistentVolumeSource{
+							Driver:       "csi-driver.example.com",
+							VolumeHandle: "existingVolumeName",
+						},
+					},
+				},
+			},
+			expected: component.NewSummary("Configuration", []component.SummarySection{
+				{
+					Header:  "CSI",
+					Content: component.NewText("{\"driver\":\"csi-driver.example.com\",\"volumeHandle\":\"existingVolumeName\"}"),
+				},
+			}...),
+		},
+	}
+
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			var sections component.SummarySections
+			sections, err := printPersistentVolumeSource(&tc.persistentVolume, sections)
+			require.NoError(t, err)
+
+			got := component.NewSummary("Configuration", sections...)
+
+			component.AssertEqual(t, tc.expected, got)
+		})
+	}
+}

--- a/internal/printer/persistentvolumeclaim_test.go
+++ b/internal/printer/persistentvolumeclaim_test.go
@@ -22,7 +22,7 @@ import (
 	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
 )
 
-func Test_PersistentVolumeListHandler(t *testing.T) {
+func Test_PersistentVolumeClaimListHandler(t *testing.T) {
 	labels := map[string]string{
 		"foo": "bar",
 	}

